### DISCUSSION TITLE
23.12.05 데일리 스크럼

### DISCUSSION THREAD

**Discussion by minju412** [petit-a-petit/MIML#82] [Daily Scrum]
.

**Comment by minju412** [comment]
- 오늘은 어떤 일을 했나요?
    - 친구 요청 보낸 회원 조회 기능 테스트 작성
    - 조회 테스트에 mocking을 사용해도 의미가 있을까..? 🤔
- 내일은 어떤 일을 하나요?
    - 나에게 친구 요청 보낸 회원 조회 기능 테스트 작성

**Reply by minju412** [reply]
탈퇴한 회원이 보낸 친구 신청이 조회되는지? (현재 앱에는 탈퇴 기능 없다..)

**Reply by Nine-JH** [reply]
탈퇴 회원은 안보여줘도 될 것 같긴하네요

**Reply by minju412** [reply]
- 현재는 조회 테스트가 의미 없을 것 같지만, 같은 메서드 내에 로직이 추가되거나 변경되는 상황을 고려한다면 테스트를 작성해두는게 좋을 것 같다. 
- repository를 mocking하는 것 자체도, repository를 제대로 호출되는지 확인할수 있기 때문에 의미가 아예 없지는 않다.

**Comment by Jeonwoojae** [comment]
- 오늘은 어떤 일을 했나요?
  - 알림 기능 테스트 코드 수정 실패😢
    - @Transactional로 롤백하려고 했으나 해당 어노테이션 사용 시 spring event가 테스트에서 skip
    - 현재 방식 CompletableFuture로 테스트에서도 비동기 작업 수행 후 완료까지 대기
    - 이전 방식 executor 설정 변경으로 테스트 환경에서만 비동기를 동기처럼
- 내일은 어떤 일을 하나요?
  - 테스트 케이스 정리
  - 음악 저장 기능 테스트 작성

**Reply by Jeonwoojae** [reply]
레포지토리를 사용해서 클래스 내부에 선언한 객체들 직접 삭제하는 걸로 변경 예정입니다!
